tests/gbs/test_subgraph.py

```diff
@@ -14,113 +14,239 @@
 r"""
 Unit tests for strawberryfields.gbs.subgraph
 """
-# pylint: disable=no-self-use,unused-argument
+# pylint: disable=no-self-use,unused-argument,protected-access,redefined-outer-name
```

Comment:
- `protected-access` is needed to test those functions (ones with a leading underscore)
- `redefined-outer-name` is needed because we use the `process_planted` fixture

Reply:
Note that it might be worth adding a comment, i.e.
```python
# needed to test private functions pylint: disable=protected-access
# needed because we use the process_planted fixture pylint: disable=redefined-outer-name
```
This way it's in the source code in case a developer in future is wondering why they are required
(I _think_ that's the syntax for including comments, but it could be some permutation on that).

Reply:
@josh146 agreed it makes sense to document in code the justification for pylint disables!

Although, it seems like there are some "well-established" disable settings specifically for testing modules, e.g. `protected-access`, `no-self-use` etc. Do you think we should comment for them also?

Reply:
> Do you think we should comment for them also?

Nope, just the ones you mentioned above. It should be fine from context why `no-self-use` etc. is used

Reply:
Done 😄